### PR TITLE
RavenDB-11800 Don't allow adding node to the cluster when the leader initialize with port 0

### PR DESCRIPTION
**Pull request by aviv86**
https://issues.hibernatingrhinos.com/issue/RavenDB-11800

### CODE REVIEW

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 1]
```diff
@@ -186,6 +186,7 @@ public long LastRaftCommitIndex
         private PoolOfThreads.LongRunningWork _clusterMaintenanceSetupTask;
         private PoolOfThreads.LongRunningWork _updateTopologyChangeNotification;
 
+        public bool ValidateNotRandomPort = true;
```
Comment (ayende) [inline src/Raven.Server/ServerWide/ServerStore.cs]:
```suggestion
        public bool ValidateNonRandomPort = true;
```

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 2]
```diff
@@ -417,6 +418,18 @@ public ClusterTopology GetClusterTopology(TransactionOperationContext context)
...
         {
+
+            if (ValidateNotRandomPort)
+            {
+                if (Configuration.Core.ServerUrls != null &&
+                    Uri.TryCreate(Configuration.Core.ServerUrls[0], UriKind.Absolute, out var uri) &&
+                    uri.Port == 0)
+                {
+                    throw new InvalidOperationException($"Failed to add node '{nodeUrl}' to cluster. " +
+                                                        "Adding nodes to cluster is forbidden when the leader has port '0' in 'Configuration.Core.ServerUrls' setting");
```
Comment (ayende) [inline src/Raven.Server/ServerWide/ServerStore.cs]:
```suggestion
                                                        "Adding nodes to cluster is forbidden when the leader has port '0' in 'Configuration.Core.ServerUrls' setting. Define a fixed port for the node to enable cluster creation.");
```

**Review by ayende — COMMENTED** [review]
Also, what about the other side, so we can't add a node that was started with port 0?

**Comment by karmeli87** [comment]
Not sure if you can tell that other side started with port 0..
What we can do is to reject requests to add us to a cluster, so the only valid cluster for starting with port 0 will be single node.

**Comment by ayende** [comment]
We can give that indication in `RachisAdminHandler.GetNodeInfo`, and check that in `AddNode`

File: src/Raven.Server/ServerWide/ServerStore.cs [hunk 3]
```diff
@@ -415,8 +417,20 @@ public ClusterTopology GetClusterTopology(TransactionOperationContext context)
             return _engine.GetTopology(context);
         }
 
+        public bool HasFixedPort =>
+            Configuration.Core.ServerUrls == null ||
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/ServerStore.cs:421]: can this be `null`? thought that by default it is `localhost:8080`?

**Review by ppekrol — APPROVED** [review]

**Pull request merged**